kyo-prelude/shared/src/main/scala/kyo2/kernel/Safepoint.scala

```diff
@@ -176,7 +177,7 @@ object Safepoint:
     // TODO compiler crash if private[kyo2]
     def insertTrace(cause: Throwable)(using self: Safepoint): Unit =
         val size = Math.min(self.traceIdx, maxTraceFrames)
-        if size > 0 then
+        if size > 0 && !cause.isInstanceOf[NoStackTrace] then
```

Comment:
Do we want to have an `else` to capture a single Frame with the Throwable's name (even if it doesn't have a stack trace)? Not sure if that's already captured somewhere else...

Reply:
The `Throwable` instance already has its type/name. This condition only avoids mutating the stack trace to insert Kyo's frames.